src/extension.c

```diff
@@ -56,20 +56,33 @@ enum ExtensionState
 };
 
 static enum ExtensionState extstate = EXTENSION_STATE_INITIAL;
-static TransactionId drop_transaction_id = InvalidTransactionId;
 
 bool
 extension_is_being_dropped(Oid relid)
 {
+	/*
+	 * during a extension drop, the extension proxy table will be dropped.
+	 * This will automatically cause a relcache invalidation on the proxy
+	 * tables's OID.
+	 */
 	return relid == extension_proxy_oid;
 }
 
+
+static bool
+proxy_table_exists()
```

Comment:
I think that's safer than trying to catch every ddl operation that can lead to that because we might always miss some.